kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -146,6 +146,16 @@ define([
                 });
             });
 
+            fieldWidget.bus.on('validation', (message) => {
```

Comment:
When an input widget (specifically a text field - in the test cases, the float and int fields) gets modified, it gets validated internally. If it's invalid, the model doesn't get updated, but visually it's still there. It does send a `validation` message up the tree, though. This catches that and invalidates the whole parameter set if so.

For example, if I were to make a completely valid and ready to run set of parameters, then change a numerical field from "0" to, say, "abc", that should raise issues and invalidate the run.

Instead, what's happening right now is the input widget just says "nope, that's an invalid value, and I'm not going to save it", but leaves it in the text field. So it looks ok, but really isn't. Catching the validation message at the `filePathWidget` (and `paramsWidget`) level and propagating it on up to the `configure` widget gets us around that.

I'm not really sure if this is the best behavior - I kinda feel like the user should be able to put in and serialize whatever they value they want. If it's wrong, we should just raise an error and not let them run the cell. But this solves that little hiccup for now.

Reply:
Yeah - I saw that behaviour when I was cleaning up the tests for some of the input widgets. 😕